### PR TITLE
Nylas availability: mobile view

### PR DESCRIPTION
**Pull request by philrenaud**
Lets the availability component (and therefore scheduler component) gracefully reduce the number of dates shown when browser width is reduced, such as on mobile.

squishy-avail

# License

I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: components/availability/src/Availability.svelte [hunk 1]
```diff
@@ -254,12 +255,32 @@
...
+  $: datesToShow =
+    optimalDatesToShow < dates_to_show || show_as_week
+      ? optimalDatesToShow
+      : dates_to_show;
+
+  $: tooSmallForWeek = show_as_week && optimalDatesToShow < 7;
+
+  //#endregion layout
+
+  //#region data generation
```
Comment (philrenaud) [inline components/availability/src/Availability.svelte:280]: side-effect: I've broken up the `layout` region into layout and `data generation` for easier compartmentalizing / later modules.

File: components/availability/src/Availability.svelte [hunk 2]
```diff
@@ -254,12 +255,32 @@
...
+  let datesToShow: number;
+
+  $: optimalDatesToShow =
+    Math.floor(dayContainerWidth / MINIMUM_DAY_WIDTH) || 1;
+  $: datesToShow =
+    optimalDatesToShow < dates_to_show || show_as_week
+      ? optimalDatesToShow
+      : dates_to_show;
+
+  $: tooSmallForWeek = show_as_week && optimalDatesToShow < 7;
```
Comment (heathergh) [inline components/availability/src/Availability.svelte:276]: I'm a little confused about what this variable is

Comment (philrenaud) [reply components/availability/src/Availability.svelte:276]: this is true/false depending on whether the screen's too small to support the entire week worth of dates

File: components/availability/src/Availability.svelte [hunk 3]
```diff
@@ -269,13 +290,14 @@
     generateDayRange(
       startDay, // TODO: weird just to get show_weekends passed in
       show_as_week
-        ? timeDay.offset(startDay, 6)
-        : timeDay.offset(startDay, dates_to_show - 1),
+        ? timeDay.offset(startDay, tooSmallForWeek ? datesToShow - 1 : 6)
+        : timeDay.offset(startDay, datesToShow - 1),
```
Comment (pooja169usp) [inline components/availability/src/Availability.svelte:294]: We're following snake_case for props and camelCase for local variables. Cool 👍🏽

**Review by AaronDDM — APPROVED** [review]

**Review by pooja169usp — APPROVED** [review]
Looks & works great! 🎉

**Review by heathergh — APPROVED** [review]

**Comment by heathergh** [comment]
Nice work! I think the only improvement would be changing the styling for mobile (less than 500px) as the `section.booker` element takes up most of the screen in that view
Screen Shot 2021-09-24 at 10 34 13 AM

**Comment by philrenaud** [comment]
> Nice work! I think the only improvement would be changing the styling for mobile (less than 500px) as the `section.booker` element takes up most of the viewport

Great suggestion! We're keeping scheduler and availability as separate on purpose for exactly this reason: users can position and style these however they want to. Most of this handling would currently be done in the scheduler/index.html demo, in fact:

CleanShot 2021-09-24 at 11 19 55

**Comment by heathergh** [comment]
> > Nice work! I think the only improvement would be changing the styling for mobile (less than 500px) as the `section.booker` element takes up most of the viewport
> 
> Great suggestion! We're keeping scheduler and availability as separate on purpose for exactly this reason: users can position and style these however they want to. Most of this handling would currently be done in the scheduler/index.html demo, in fact:
> 
> CleanShot 2021-09-24 at 11 19 55

Noted! Then I'll scratch that suggestion :)

**Pull request merged**